### PR TITLE
[clippy] Fix clippy issues for crate sp-core

### PR DESCRIPTION
**Pull request by adoerr**
Fix `clippy` issues for crate `sp-core`.  Running 

```
cargo +nightly clippy --all-targets -- -D warnings --no-deps
```
for the crate no longer produces any errors. 

I did allow some clippy checks, where a change would have impacted the function/method signature.

### CODE REVIEW

File: primitives/core/benches/bench.rs [hunk 1]
```diff
@@ -12,6 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#![allow(deprecated)]
```
Comment (bkchr) [inline primitives/core/benches/bench.rs:15]:
```suggestion
```

File: primitives/core/src/ed25519.rs [hunk 2]
```diff
@@ -522,10 +522,7 @@ impl TraitPair for Pair {
 			Err(_) => return false
 		};
 
-		match public_key.verify(message.as_ref(), &sig) {
-			Ok(_) => true,
-			_ => false,
-		}
+		matches!(public_key.verify(message.as_ref(), &sig), Ok(_))
```
Comment (bkchr) [inline primitives/core/src/ed25519.rs:525]:
```suggestion
		public_key.verify(message.as_ref(), &sig).is_ok()
```

File: primitives/core/src/lib.rs [hunk 3]
```diff
@@ -285,15 +285,15 @@ pub trait TypeId {
 #[derive(Encode, Decode, PassByEnum, Copy, Clone)]
 pub enum LogLevel {
 	/// `Error` log level.
-	Error = 1,
+	Error = 1_isize,
```
Comment (bkchr) [inline primitives/core/src/lib.rs:288]:
```suggestion
	Error = 1u8,
```

?

File: primitives/core/src/lib.rs [hunk 4]
```diff
@@ -340,17 +340,17 @@ impl From<LogLevel> for log::Level {
 #[derive(Encode, Decode, PassByEnum, Copy, Clone)]
 pub enum LogLevelFilter {
 	/// `Off` log level filter.
-	Off = 0,
+	Off = 0_isize,
```
Comment (bkchr) [inline primitives/core/src/lib.rs:343]: Same

File: primitives/core/src/offchain/mod.rs [hunk 5]
```diff
@@ -66,12 +66,12 @@ pub enum StorageKind {
 	/// that is re-run at block `N(hash2)`.
 	/// This storage can be used by offchain workers to handle forks
 	/// and coordinate offchain workers running on different forks.
-	PERSISTENT = 1,
+	PERSISTENT = 1_isize,
```
Comment (bkchr) [inline primitives/core/src/offchain/mod.rs:69]: Same

File: primitives/core/src/offchain/mod.rs [hunk 6]
```diff
@@ -108,11 +108,11 @@ impl From<HttpRequestId> for u32 {
 #[repr(C)]
 pub enum HttpError {
 	/// The requested action couldn't been completed within a deadline.
-	DeadlineReached = 1,
+	DeadlineReached = 1_isize,
```
Comment (bkchr) [inline primitives/core/src/offchain/mod.rs:111]: Same.

**Review by bkchr — CHANGES_REQUESTED** [review]
Please remove all these `clippy` attributes

File: primitives/core/src/lib.rs (hunk 3 above)
Comment (adoerr) [reply primitives/core/src/lib.rs:288]:
```
error[E0308]: mismatched types
   --> primitives/core/src/lib.rs:288:10
    |
288 |     Error = 1u8,
    |             ^^^ expected `isize`, found `u8`
    |
help: change the type of the numeric literal from `u8` to `isize`
    |
288 |     Error = 1isize,
    |             ^^^^^^
```
`u8` does not compile.

**Comment by adoerr** [comment]
You mean all `#[allow(clippy::*)]` not only the `#[allow(deprecated)]`?

Frankly this defeats the whole purpose. If we still get `clippy` errors, the remaining changes a sort of superfluous. Allow some `clippy` checks at least documents, we are aware of the issue but overrule.

**Comment by bkchr** [comment]
But we don't use clippy and also don't want to have these attributes in the code.

File: primitives/core/src/lib.rs (hunk 3 above)
Comment (bkchr) [reply primitives/core/src/lib.rs:288]: Hmm okay.

File: primitives/core/benches/bench.rs [hunk 7]
```diff
@@ -32,16 +32,18 @@ fn get_key(key_size: u32) -> Vec<u8> {
 	let mut rnd = rnd.iter().cycle();
 
 	(0..key_size)
-		.map(|_| rnd.next().unwrap().clone())
+		.map(|_| *rnd.next().unwrap())
 		.collect()
 }
 
+#[allow(clippy::clone_on_copy, clippy::ptr_arg)]
```
Comment (bkchr) [inline primitives/core/benches/bench.rs:39]: There are still attributes

**Review by KiChjang — APPROVED** [review]

**Review by bkchr — APPROVED** [review]

**Pull request merged**